src/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.ts

```diff
@@ -193,10 +196,35 @@ export class TextSearchQuickAccess extends PickerQuickAccessProvider<IPickerQuic
 					},
 				});
 			}
+		}
+		return picks;
+	}
+	protected _getPicks(contentPattern: string, disposables: DisposableStore, token: CancellationToken): FastAndSlowPicks<IQuickPickItem> {
+
+		const allMatches = this.doSearch(contentPattern);
```

Comment:
You should absolutely be passing down the CancellationToken passed in to `_getPicks` as far as you can in the async stuff.

That token will cancel when the user starts typing again and will help keep things snappy.

Reply:
thanks for the catch! I definitely meant to do that, just forgot to make the final connections